src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -72,6 +72,18 @@ extends AbstractMap[A, B]
     else Some(e.value)
   }
 
+  override def getOrElseUpdate(key: A, defaultValue: => B): B = {
+    val i = index(elemHashCode(key))
+    val entry = findEntry0(key, i)
+    if (entry != null)
+      entry.value
+    else {
+      val newEntry = createNewEntry(key, defaultValue)
+      addEntry0(newEntry, i)
+      newEntry.value
+    }
+  }
```

Comment:
The idea is very good, but we don't have a `value` in `HashTable`'s `Entry`, only `key` and `next`.

The other alternative would be to call `findOrAddEntry` directly, but then `defaultValue` would be evaluated twice, which should probably be avoided, right?

```scala
override def getOrElseUpdate(key: A, defaultValue: => B): B = {
  val e = findOrAddEntry(key, defaultValue)
  if (e ne null) e.value else defaultValue
}
```